transformer_engine/common/include/transformer_engine/comm_gemm.h

```diff
@@ -0,0 +1,75 @@
+/*************************************************************************
+ * Copyright (c) 2022-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#ifndef TRANSFORMER_ENGINE_COMMON_COMM_GEMM_H_
+#define TRANSFORMER_ENGINE_COMMON_COMM_GEMM_H_
+
+#include <cublasmp.h>
+#include <nccl.h>
+#include <stdint.h>
+
+#include "transformer_engine.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct CommGemmCtx CommGemmCtx;
+
+/*! \brief Create a comm-gemm context.
+ */
```

Comment:
Should have parameters etc. listed and described. Also, I don't quite like the "comm-gemm" name - if you do not know what this is about it would not tell you anything on its own. Maybe collective gemm would be better?